File: core/bloom/messages/src/notes.rs
use super::{from_message, Message};
use bloom_models::notes;
use serde::{Deserialize, Serialize};

////////////////////////////////////////////////////////////////////////////////////////////////////
/// GUI
////////////////////////////////////////////////////////////////////////////////////////////////////

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct GuiListNotes {}
from_message!(GuiListNotes, Message::NotesGuiListNotes);

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct GuiNotes {
    pub notes: Vec<notes::db::Note>,
}
from_message!(GuiNotes, Message::NotesGuiNotes);

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct GuiDeleteNote {
    pub id: String,
}
from_message!(GuiDeleteNote, Message::NotesGuiDeleteNote);

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct GuiGetArchive {}
from_message!(GuiGetArchive, Message::NotesGuiGetArchive);

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct GuiCreateNote {
    pub title: String,
    pub body: String,
    pub color: i64,
}
from_message!(GuiCreateNote, Message::NotesGuiCreateNote);

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct GuiNote {
    pub note: notes::db::Note,
}
from_message!(GuiNote, Message::NotesGuiNote);

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct GuiUpdateNote {
    pub note: notes::db::Note,
}
from_message!(GuiUpdateNote, Message::NotesGuiUpdateNote);

File: desktop/src/native/messages/notes.ts
/* eslint-disable camelcase */

export interface Note {
  id: string,
  title: string,
  body: string,
  created_at: Date,
  updated_at: Date,
  color: number,
  archived_at: Date | null,
  is_pinned: boolean,
}

export interface GuiNote {
  note: Note,
}

export interface GuiNotes {
  notes: Note[],
}

export interface GuiCreateNote {
  title: string,
  body: string,
  color: number,
}

export interface GuiDeleteNote {
  id: string,
}
